src/WebJobs.Script/Diagnostics/WorkerLogger.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Extensions.Logging;
+using Microsoft.Azure.WebJobs.Script.Extensions;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    /// <summary>
+    /// A wrapper class to allow ILoggers from functions to write to the existing FileTraceWriter. This allows
+    /// logs to show up in the log files and stream in the portal.
+    /// </summary>
+    internal class WorkerLogger : ILogger
+    {
+        private readonly Func<string, LogLevel, bool> _filter;
+        private readonly string _categoryName;
+        private readonly string _language;
+        private readonly string _workerId;
+        private readonly IFunctionTraceWriterFactory _traceWriterFactory;
+
+        public WorkerLogger(string categoryName, IFunctionTraceWriterFactory traceWriterFactory, Func<string, LogLevel, bool> filter)
+        {
+            _categoryName = categoryName;
+            _filter = filter;
+            _traceWriterFactory = traceWriterFactory;
+            var parts = categoryName.Split('.');
+            _language = parts[1];
+            _workerId = parts[2];
+        }
+
+        public static Regex Regex { get; } = new Regex(@"^Worker\.[^\s]+\.[^\s]+");
```

Comment:
Rather than expose a regex for this, how about an IsRpcCategory(string category)? (Where Rpc is whatever term we come up with)